File: lib/statsailr_procs_base/proc_setting/proc_common/dev_copy.rb
module DevCopySetting
  include ProcSettingModule
  source_r_file(__dir__, File.basename(__FILE__ , ".rb") + ".R")

  def setting_for_dev_copy( setting )
    setting.libname = nil
    setting.envname = "sts_dev_copy"
    setting.func_name = "dev_copy"
    setting.main_arg_and_how_to_treat = [ "device" , :read_as_symbol , :allow_nil ]
    setting.runtime_args = {}
    setting.store_result = false
    setting.print_opt = false
  end
end

File: lib/statsailr_procs_base/proc_setting/proc_common/dev_copy.R
STS_RC_DevCopy = setRefClass("STS_RC_DevCopy")
STS_RC_DevCopy$methods(
  dev_copy = function( device, ... ){
    dev.copy( device , ...)
    dev.off()
  }
)

sts_dev_copy = STS_RC_DevCopy()
